cli/cmd/up/up.go

```diff
@@ -55,6 +55,21 @@ func (u *Up) Run(c *clicontext.CLIContext) error {
 		return err
 	}
 
+	rev := ""
+	if stack.Spec.Build != nil {
+		rev = stack.Spec.Build.Revision
+	}
+
+	if rev == "" {
+		rev = stack.Status.Revision
+	}
+
+	if answer == nil {
+		answer = map[string]string{}
+	}
+	answer["REVISION"] = rev
+	answer["NAMESPACE"] = c.GetSetNamespace()
```

Comment:
Can you move your logic in here? https://github.com/rancher/rio/pull/895/commits/c01137861ee2fa6ae26971242870a16c95de2327#diff-d4629a579ebc0e6d62bf1a5baefc2608R132